mangaki/mangaki/admin.py

```diff
@@ -229,60 +230,103 @@ def make_nsfw(self, request, queryset):
 
     make_nsfw.short_description = "Rendre NSFW les œuvres sélectionnées"
 
-    # FIXME : https://github.com/mangaki/mangaki/issues/205
     def update_tags_via_anidb(self, request, queryset):
-        if request.POST.get("post"):
-            kept_ids = request.POST.getlist('checks')
-            for anime_id in kept_ids:
-                anime = Work.objects.get(id=anime_id)
-
-                retrieved_tags = anime.retrieve_tags(client)
-                deleted_tags = retrieved_tags["deleted_tags"]
-                added_tags = retrieved_tags["added_tags"]
-                updated_tags = retrieved_tags["updated_tags"]
-
-                tags = deleted_tags
-                tags.update(added_tags)
-                tags.update(updated_tags)
-
-                client.update_tags(anime, tags)
+        works = queryset.all()
 
-            self.message_user(request, "Modifications sur les tags faites")
+        if request.POST.get('confirm'): # Updating tags has been confirmed
+            to_update_work_ids = set(list(map(int, request.POST.getlist('to_update_work_ids'))))
+            nb_updates = len(to_update_work_ids)
+
+            work_ids = list(map(int, request.POST.getlist('work_ids')))
```

Comment:
AFAIK, `list` is unnecessary here because you only want to iterate on the data, not to store it somewhere.